### PR TITLE
DATAACCESS: Create ProductDetailRepository, ProductDetailFashionRepository, ProductCommentRepository and fix ISearchable

### LINKED ISSUE #37: DATAACCESS : Create ProductDetailFashionRepository

### LINKED ISSUE #38: DATAACCESS : Create ProductDetailRepository

### PR DESCRIPTION
**Pull request by HasanboyXomidov**
Close #38 
Close #37

### CODE REVIEW

File: src/Eclo.DataAccess/Repositories/ProductDetailFashionRepository.cs [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+    public async Task<long> CountAsync()
+    {
+        try
+        {
+            await _connection.OpenAsync();
+            string query = $"select count(*) from product_detail_fashions";
+            var result = await _connection.QuerySingleAsync<long>(query);
+
+            return result;
+
```
Comment (ShohruhUzDev) [inline src/Eclo.DataAccess/Repositories/ProductDetailFashionRepository.cs:20]: delete empty line

File: src/Eclo.DataAccess/Repositories/ProductDetailFashionRepository.cs [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+    }
+
+    public async Task<int> CreateAsync(ProductDetailFashion entity)
+    {
+        try
+        {
+            await _connection.OpenAsync();
+            string query = "INSERT INTO public.product_detail_fashions(product_detail_id, image_path, created_at, " +
+                "updated_at) VALUES (@ProductDetailId,@ImagePath,@CreatedAt,@UpdatedAt);";
+            var result = await _connection.ExecuteAsync(query, entity);
```
Comment (ShohruhUzDev) [inline src/Eclo.DataAccess/Repositories/ProductDetailFashionRepository.cs:39]: put empty line

File: src/Eclo.DataAccess/Repositories/ProductDetailFashionRepository.cs [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+        }
+    }
+
+    public async Task<int> DeleteAsync(long id)
+    {
+        try
+        {
+            await _connection.OpenAsync();
+            string query = "DELETE FROM public.product_detail_fashions " +
+                "WHERE id=@Id;";
```
Comment (ShohruhUzDev) [inline src/Eclo.DataAccess/Repositories/ProductDetailFashionRepository.cs:60]: bir qatorda bulaversin

File: src/Eclo.DataAccess/Repositories/ProductDetailFashionRepository.cs [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+            var result = await _connection.ExecuteAsync(query, new { Id = id });
+
+            return result;
+
```
Comment (ShohruhUzDev) [inline src/Eclo.DataAccess/Repositories/ProductDetailFashionRepository.cs:64]: delete empty line

File: src/Eclo.DataAccess/Repositories/ProductDetailFashionRepository.cs [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+    }
+
+    public async Task<IList<ProductDetailFashionViewModel>> GetAllAsync(PaginationParams @params)
+    {
+        try
+        {
+            await _connection.OpenAsync();
+            string query = $"SELECT * FROM public.product_detail_fashion_view order by id desc " +
+                $"offset {@params.GetSkipCount()} limit {@params.PageSize}";
+            var result = (await _connection.QueryAsync<ProductDetailFashionViewModel>(query)).ToList();
```
Comment (ShohruhUzDev) [inline src/Eclo.DataAccess/Repositories/ProductDetailFashionRepository.cs:83]: ikki qatorli kod boshqalarida ajralib tursin

File: src/Eclo.DataAccess/Repositories/ProductDetailFashionRepository.cs [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+            await _connection.CloseAsync();
+        }
+    }
+
+    public async Task<ProductDetailFashionViewModel?> GetByIdAsync(long id)
+    {
+        try
+        {
+            await _connection.OpenAsync();
+            string query = $"SELECT * FROM public.product_detail_fashion_view where id=@Id";
```
Comment (ShohruhUzDev) [inline src/Eclo.DataAccess/Repositories/ProductDetailFashionRepository.cs:103]: put empty line

**Review by ShohruhUzDev — APPROVED** [review]

File: src/Eclo.DataAccess/Repositories/ProductDetailFashionRepository.cs (hunk 6 above)
Comment (ShohruhUzDev) [reply src/Eclo.DataAccess/Repositories/ProductDetailFashionRepository.cs:103]: qolgan kodlar ham shu tarzda tugirlansin. Hamasini aytmadim.

**Review by sodiqjonovich — DISMISSED** [review]
ok

**Review by javohirtohirovich — APPROVED** [review]

**Review by Maxsetovich — APPROVED** [review]
Created ProductDetailRepository, ProductDetailFashionRepository, ProductCommentRepository and fix ISearchable

**Pull request merged**